internal/datasystem/fdv1_datasystem.go

```diff
@@ -0,0 +1,146 @@
+	dataStoreStatusProvider     interfaces.DataStoreStatusProvider
+	flagChangeEventBroadcaster  *internal.Broadcaster[interfaces.FlagChangeEvent]
+	dataStore                   subsystems.DataStore
+	dataSource                  subsystems.DataSource
+}
+
+func NewFDv1(dataStoreFactory subsystems.ComponentConfigurer[subsystems.DataStore], dataSourceFactory subsystems.ComponentConfigurer[subsystems.DataSource], clientContext *internal.ClientContextImpl) (*FDv1, error) {
+	system := &FDv1{
+		dataSourceStatusBroadcaster: internal.NewBroadcaster[interfaces.DataSourceStatus](),
+		dataStoreStatusBroadcaster:  internal.NewBroadcaster[interfaces.DataStoreStatus](),
+		flagChangeEventBroadcaster:  internal.NewBroadcaster[interfaces.FlagChangeEvent](),
+	}
+
+	dataStoreUpdateSink := datastore.NewDataStoreUpdateSinkImpl(system.dataStoreStatusBroadcaster)
+	storeFactory := dataStoreFactory
+	if storeFactory == nil {
+		storeFactory = ldcomponents.InMemoryDataStore()
+	}
+	clientContextWithDataStoreUpdateSink := clientContext
+	clientContextWithDataStoreUpdateSink.DataStoreUpdateSink = dataStoreUpdateSink
+	store, err := storeFactory.Build(clientContextWithDataStoreUpdateSink)
+	if err != nil {
+		return nil, err
+	}
+	system.dataStore = store
+
+	system.dataStoreStatusProvider = datastore.NewDataStoreStatusProviderImpl(store, dataStoreUpdateSink)
+
+	dataSourceUpdateSink := datasource.NewDataSourceUpdateSinkImpl(
+		store,
+		system.dataStoreStatusProvider,
+		system.dataSourceStatusBroadcaster,
+		system.flagChangeEventBroadcaster,
+		clientContext.GetLogging().LogDataSourceOutageAsErrorAfter,
+		clientContext.GetLogging().Loggers,
+	)
+
+	dataSource, err := createDataSource(clientContext, dataSourceFactory, dataSourceUpdateSink)
+	if err != nil {
+		return nil, err
+	}
+	system.dataSource = dataSource
+	system.dataSourceStatusProvider = datasource.NewDataSourceStatusProviderImpl(
+		system.dataSourceStatusBroadcaster,
+		dataSourceUpdateSink,
+	)
+
+	return system, nil
+
+}
+
+func createDataSource(
+	context *internal.ClientContextImpl,
+	dataSourceBuilder subsystems.ComponentConfigurer[subsystems.DataSource],
+	dataSourceUpdateSink subsystems.DataSourceUpdateSink,
+) (subsystems.DataSource, error) {
+	if context.Offline {
+		context.GetLogging().Loggers.Info("Starting LaunchDarkly client in offline mode")
+		dataSourceUpdateSink.UpdateStatus(interfaces.DataSourceStateValid, interfaces.DataSourceErrorInfo{})
+		return datasource.NewNullDataSource(), nil
+	}
+	factory := dataSourceBuilder
+	if factory == nil {
+		// COVERAGE: can't cause this condition in unit tests because it would try to connect to production LD
+		factory = ldcomponents.StreamingDataSource()
+	}
+	contextCopy := *context
+	contextCopy.BasicClientContext.DataSourceUpdateSink = dataSourceUpdateSink
+	return factory.Build(&contextCopy)
+}
+
+func (f *FDv1) DataSourceStatusBroadcaster() *internal.Broadcaster[interfaces.DataSourceStatus] {
```

Comment:
This series of getters is to satisfy the SDK client's public interface, where these status providers are exposed. The FDv2 data system will also need to return something here, but they may not be functional (for now, or maybe I'll find a way to make them function.)